### PR TITLE
feat: add order notifications metrics

### PR DESCRIPTION
**Pull request by rileydcampbell**
## Summary
- Add order notification metrics 
- Add order notification alarms on metrics

### CODE REVIEW

File: lib/handlers/order-notification/handler.ts [hunk 1]
```diff
@@ -54,11 +54,20 @@ export class OrderNotificationHandler extends DynamoStreamLambdaHandler<Containe
...
-          result.status == 'fulfilled' && result?.value?.status >= 200 && result?.value?.status <= 202
-            ? log.info({ result: result.value }, 'Success: New order record sent to registered webhook.')
-            : failedRequests.push(result)
-        )
+
+        results.forEach((result, index) => {
+          metrics.putMetric(`OrderNotificationAttempt-chain-${newOrder.chainId}`, 1)
+          if (result.status == 'fulfilled' && result?.value?.status >= 200 && result?.value?.status <= 202) {
+            log.info(
+              { result: result.value },
```
Comment (zhongeric) [inline lib/handlers/order-notification/handler.ts:62]: Don't think we need a log for success case

File: test/handlers/order-notification.test.ts [hunk 2]
```diff
@@ -95,7 +95,11 @@ describe('Testing new order Notification handler.', () => {
     expect(logInfoMock).toBeCalledTimes(2)
     expect(logInfoMock).toBeCalledWith(
       { result: { status: 200 } },
-      'Success: New order record sent to registered webhook.'
+      'Success: New order record sent to registered webhook webhook.com/1.'
```
Comment (zhongeric) [inline test/handlers/order-notification.test.ts:98]: should add a test with an axios mock that throws with 4XX

File: lib/handlers/order-notification/handler.ts (hunk 1 above)
Comment (rileydcampbell) [reply lib/handlers/order-notification/handler.ts:62]: It's been helpful for debugging and just making sure the integrators are sending / receiving the proper data.

File: lib/handlers/order-notification/handler.ts [hunk 3]
```diff
@@ -54,11 +54,20 @@ export class OrderNotificationHandler extends DynamoStreamLambdaHandler<Containe
...
+              `Success: New order record sent to registered webhook ${registeredEndpoints[index].url}.`
+            )
+            metrics.putMetric(`OrderNotificationSendSuccess-chain-${newOrder.chainId}`, 1)
+          } else {
+            failedRequests.push(result)
```
[the remark this answers is not in the document]
Comment (rileydcampbell) [reply lib/handlers/order-notification/handler.ts:67]: So I added all the failed requests to an array then on line 73 I log all the details for the failed notifications.

File: lib/handlers/order-notification/handler.ts [hunk 4]
```diff
@@ -54,11 +54,20 @@ export class OrderNotificationHandler extends DynamoStreamLambdaHandler<Containe
...
         const results = await Promise.allSettled(requests)
-        results.forEach((result) =>
-          result.status == 'fulfilled' && result?.value?.status >= 200 && result?.value?.status <= 202
-            ? log.info({ result: result.value }, 'Success: New order record sent to registered webhook.')
-            : failedRequests.push(result)
-        )
+
+        results.forEach((result, index) => {
+          metrics.putMetric(`OrderNotificationAttempt-chain-${newOrder.chainId}`, 1)
+          if (result.status == 'fulfilled' && result?.value?.status >= 200 && result?.value?.status <= 202) {
```
[the remark this answers is not in the document]
Comment (rileydcampbell) [reply lib/handlers/order-notification/handler.ts:60]: Yep it does. So I noticed fillers were sending OK responses as anything >= 200 and <= 202. Then if they 5xx this promise will resolve as `rejected` and I'll log that down below.

Also, [here's](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) how `Promise.allSettled` works. I didn't know how it worked until I wrote this but pretty sweet for use cases where the results of one network requests shouldn't effect the others.

File: bin/stacks/lambda-stack.ts [hunk 5]
```diff
@@ -335,5 +337,53 @@ export class LambdaStack extends cdk.NestedStack {
...
     }
+
+    // TODO: add more chains when we support them
+    const orderNotificationErrorRateMetric = new MathExpression({
+      expression: '100*(errors/attempts)',
+      period: Duration.minutes(5),
+      usingMetrics: {
+        errors: new Metric({
+          namespace: 'Uniswap',
+          metricName: 'OrderNotificationSendFailure-chain-1',
```
[the remark this answers is not in the document]
Comment (rileydcampbell) [reply bin/stacks/lambda-stack.ts:348]: Yep yep, just added alarms for each chain. Looks like what we've been doing in this repo is `-chain-${}`.

File: lib/handlers/order-notification/handler.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (zhongeric) [reply lib/handlers/order-notification/handler.ts:60]: oh sweet that is pretty cool

**Review by zhongeric — APPROVED** [review]

**Pull request merged**